### ISSUE TITLE
[java] LocalVariableCouldBeFinal false-positive with lombok.val

### ISSUE THREAD

**Issue by Miuler** [pmd/pmd#5079]
**Rule:LocalVariableCouldBeFinal**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#methodargumentcouldbefinal>

**Description: lombok.val is equivalet to final class**

**Code Sample demonstrating the issue:**

```java
import lombok.val;

val test04 = logicTestService.test();
// This is equivalent to:
// final String test04 = logicTestService.test();
```

**Error:**

LocalVariableCouldBeFinal:      Local variable 'test04' could be declared final
